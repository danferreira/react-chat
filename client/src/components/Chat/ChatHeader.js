import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Avatar from '../Avatar/Avatar'

const ChatHeaderWrapper = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    min-height: 10vh;
    background: #353838;
    border-bottom: solid 1px #e8e8e8;
`;

const ChatName = styled.span`
    margin: 10px;
    color: #fff;
`

const propTypes = {
    contact: PropTypes.shape({
        name: PropTypes.string.isRequired,
        image: PropTypes.string
    }).isRequired
}

const ChatHeader = ({ contact }) => (
    <ChatHeaderWrapper>
        <Avatar
            rounded
            source={contact.image}
        />
        <ChatName>{contact.name}</ChatName>
    </ChatHeaderWrapper>
);

ChatHeader.propTypes = propTypes;

export default ChatHeader;