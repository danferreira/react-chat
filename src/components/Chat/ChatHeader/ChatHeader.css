.chat-header {
    display: flex;
    justify-content: center;
    align-items: center;
    min-height: 10vh;
    background: #353838;
    border-bottom: solid 1px #e8e8e8;
}

.chat-header .name {
    margin: 10px;
    color: #fff;
}