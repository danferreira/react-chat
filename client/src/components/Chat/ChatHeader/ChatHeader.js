import React from 'react';
import PropTypes from 'prop-types';

import Avatar from '../../Avatar/Avatar'
import './ChatHeader.css';

const propTypes = {
    contact: PropTypes.shape({
        name: PropTypes.string.isRequired,
        image: PropTypes.string
    }).isRequired
}

const ChatHeader = ({ contact }) => (
    <header>
        <div className="chat-header">
            <Avatar
                rounded
                source={contact.image}
            />
            <span className="name">{contact.name}</span>
        </div>
    </header>
);

ChatHeader.propTypes = propTypes;

export default ChatHeader;