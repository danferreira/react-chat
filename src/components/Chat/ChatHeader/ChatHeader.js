import React from 'react';

import Avatar from '../../Avatar/Avatar'
import './ChatHeader.css';

const ChatHeader = ({contact}) => (
    <header>
        <div className="chat-header">
            <Avatar image={contact.image} />
            <span className="name">{contact.name}</span>
        </div>
    </header>
);

export default ChatHeader;