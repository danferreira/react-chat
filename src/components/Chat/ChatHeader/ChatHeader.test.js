import React from 'react';
import { shallow } from 'enzyme';

import ChatHeader from './ChatHeader';

describe('<ChatHeader />', () => {

    it('should render default image', () => {

        const contact = {
            name: "Dumb"
        }
        
        var wrapper = shallow(<ChatHeader contact={contact} />);
        expect(wrapper).toMatchSnapshot();
    })

    it('should render given image', () => {

        const contact = {
            name: "Dumb",
            image: "img/foo.jpg"
        }

        var wrapper = shallow(<ChatHeader contact={contact} />);
        expect(wrapper).toMatchSnapshot();
    })
});