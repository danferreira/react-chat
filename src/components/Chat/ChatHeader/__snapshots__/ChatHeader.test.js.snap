// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatHeader /> should render default image 1`] = `
<header>
  <div
    className="chat-header"
  >
    <Avatar
      rounded={true}
      size="small"
    />
    <span
      className="name"
    >
      Dumb
    </span>
  </div>
</header>
`;

exports[`<ChatHeader /> should render given image 1`] = `
<header>
  <div
    className="chat-header"
  >
    <Avatar
      rounded={true}
      size="small"
      source="img/foo.jpg"
    />
    <span
      className="name"
    >
      Dumb
    </span>
  </div>
</header>
`;
